### PR TITLE
Supertype logger improvements

### PR DESCRIPTION
**Pull request by nsaenz7**
Adding a public api to allow customization of supertype's logging functionality - now you can send in your own logger function.

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+## 3.0.1
+* Adding a public api to allow customization of supertype's logging functionality.
```
Comment (sshankar19) [inline HISTORY.md]: minor version upgrade, not 3.0.1

Comment (nsaenz7) [reply HISTORY.md]: good catch, fixed!

File: src/SupertypeLogger.ts [hunk 2]
```diff
@@ -178,7 +186,7 @@ export class SupertypeLogger {
         }
     }
 
-    sendToLog(level, json) {
+    sendToLog(level, json, ...rawData) {
```
Comment (hackerrdave) [inline src/SupertypeLogger.ts]: where are we using `rawData` in this function?

Comment (nsaenz7) [reply src/SupertypeLogger.ts]: right now, we add some additional properties to the object that's going to get logged out for convenience (formatting, adding some timestamps, etc.). Eventually, our internal logger module will take care of that. so we want to keep our existing logs formatted, while also allowing the module logger to use an untouched version of the data that came in.

Comment (hackerrdave) [reply src/SupertypeLogger.ts]: I just mean - `sendToLog` listed as an argument of the function, but its not being used in the function?

Comment (nsaenz7) [reply src/SupertypeLogger.ts]: spoke offline yesterday, to summarize - we needed this function signature to accept three args, even though our default implementation is only utilizing two. resolution: put a code comment explaining the role of that third param.

**Pull request merged**